### PR TITLE
core: rename system calls (s/tee_svc_/syscall_/)

### PR DESCRIPTION
**Pull request by jforissier**
Assign the syscall_ prefix to all system calls to make them stand out
from other functions.

Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```

File: core/arch/arm/tee/arch_svc_private.h [hunk 1]
```diff
@@ -30,9 +30,9 @@
 #include <tee_api_types.h>
 
 /* void argument but in reality it can be any number of arguments */
-typedef TEE_Result (*tee_svc_func)(void);
+typedef TEE_Result (*syscal_t)(void);
```
Comment (pascal-brand-st-dev) [inline core/arch/arm/tee/arch_svc_private.h]: s/syscal_t/syscall_t

**Comment by pascal-brand-st-dev** [comment]
Apart from syscal_t which should be syscall_t (this makes travis wrong)

    Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

File: core/arch/arm/tee/arch_svc_private.h (hunk 1 above)
Comment (jforissier) [inline core/arch/arm/tee/arch_svc_private.h]: Oops. Will fix.

**Comment by jforissier** [comment]
Typo fixed, review tags added, and commit rebased onto master.

**Pull request merged**